src/midgard/polyline2.cc

```diff
@@ -314,6 +314,31 @@ void Polyline2<coord_t>::Generalize(container_t& polyline,
     DouglasPeucker<coord_t>(polyline, epsilon_m, exclusions);
 }
 
+template <typename coord_t>
+template <typename container_t>
+typename container_t::value_type::first_type
+Polyline2<coord_t>::HausdorffDistance(const container_t& l1, const container_t& l2) {
```

Comment:
on very very large component of the failing tests on arm were floating point noise ever so slightly changing the results of tests from the expected result. this showed itself mostly in shape comparisions. to combat this ive implemented the hausdorff distance metric which lets us tell the maximum (of the minimum) distance (in meters) between two shapes